src/Common/src/CoreLib/System/BitConverter.cs

```diff
@@ -19,10 +19,23 @@ public static class BitConverter
         // This field indicates the "endianess" of the architecture.
         // The value is set to true if the architecture is
         // little endian; false if it is big endian.
+#if MONO
+		[Intrinsic]
+		public static readonly bool IsLittleEndian;
+
+		static BitConverter () {
+			unsafe {
+				ushort i = 0x1234;
```

Comment:
I think `int i = 1` would  be clearest here.

Reply:
And then just check the low  byte  for `0` or `1`. `1234` is unnecessarily cute.

Reply:
What does the instrinic attribute mean?

Reply:
It just a market which means the JIT is going to optimize it in some way.

Reply:
Are you sure the explicit static ctor won't break AOT inlining for IsLittleEndian ?